### ISSUE TITLE
A problem with  owned Type and SetValueConverter at runtime.

### ISSUE THREAD

**Issue by yangzhongke** [dotnet/efcore#25532]
I tried to implement Strongly-Typed-Id in EF Core.
The following code works well:
```csharp
public abstract record StronglyTypedId
{
	public Guid Value { get; init; }
	public StronglyTypedId()
	{
		this.Value = Guid.NewGuid();
	}
	public StronglyTypedId(Guid id)
	{
		this.Value = id;
	}
}
public record CategoryId : StronglyTypedId
{
	public CategoryId()
	{

	}

	public CategoryId(Guid id) : base(id)
	{

	}
}

public class Category : BaseEntity, IAggregateRoot, ISoftDelete
{
	private Category() { }
	public CategoryId Id { get; set; }
	public int SequenceNumber { get; private set; }
	public string Name { get; private set; }
}		

public class CategoryConfig : IEntityTypeConfiguration<Category>
{
	public void Configure(EntityTypeBuilder<Category> builder)
	{
		builder.HasKey(p => p.Id);
		builder.Property(p => p.Id)
			   .HasConversion(id => id.Value, value => new CategoryId(value));
	}
}	
```

However, when I tried to set  valueconverters for every Strongly-Typed-Id properties at runtime, the code is as following:
Step one: Remove "builder.Property(p => p.Id).HasConversion(id => id.Value, value => new CategoryId(value));" from CategoryConfig 
Step Two:
```csharp
public static class StronglyTypedIdHelper
{
	public static void AddStronglyTypedIdConversions(this ModelBuilder modelBuilder)
	{
		foreach (var entityType in modelBuilder.Model.GetEntityTypes())
		{
			if (entityType.IsOwned())
			{
				continue;
			}
			foreach (var property in entityType.GetProperties())
			{
				Type propertyType = property.ClrType;
				if (IsStronglyTypedId(propertyType))
				{                        
					Type valueType = typeof(Guid);
					var converter = StronglyTypedIdConverters.GetOrAdd(
						propertyType,
						_ => CreateStronglyTypedIdConverter(propertyType, valueType));
					property.GetValueConverter();
					property.SetValueConverter(converter);
				}
			}
		}
	}

	private static bool IsStronglyTypedId(Type type)
	{
	   return type.IsAssignableTo(typeof(StronglyTypedId));
	}

	private static readonly ConcurrentDictionary<Type, ValueConverter> StronglyTypedIdConverters = new();

	private static ValueConverter CreateStronglyTypedIdConverter(
		Type stronglyTypedIdType,
		Type valueType)
	{
		// id => id.Value
		var toProviderFuncType = typeof(Func<,>)
			.MakeGenericType(stronglyTypedIdType, valueType);
		var stronglyTypedIdParam = Expression.Parameter(stronglyTypedIdType, "id");
		var toProviderExpression = Expression.Lambda(
			toProviderFuncType,
			Expression.Property(stronglyTypedIdParam, "Value"),
			stronglyTypedIdParam);

		// value => new ProductId(value)
		var fromProviderFuncType = typeof(Func<,>)
			.MakeGenericType(valueType, stronglyTypedIdType);
		var valueParam = Expression.Parameter(valueType, "value");
		var ctor = stronglyTypedIdType.GetConstructor(new[] { valueType });
		var fromProviderExpression = Expression.Lambda(
			fromProviderFuncType,
			Expression.New(ctor, valueParam),
			valueParam);

		var converterType = typeof(ValueConverter<,>)
			.MakeGenericType(stronglyTypedIdType, valueType);

		return (ValueConverter)Activator.CreateInstance(
			converterType,
			toProviderExpression,
			fromProviderExpression,
			null);
	}
}
```
Step Three: invoke AddStronglyTypedIdConversions in OnModelCreating
``` csharp
        protected override void OnModelCreating(ModelBuilder modelBuilder)
        {
            base.OnModelCreating(modelBuilder);
            modelBuilder.ApplyConfigurationsFromAssembly(this.GetType().Assembly);
           modelBuilder.AddStronglyTypedIdConversions();
        }
```

The code above cannot work properly, and exception 'The entity type XXXId requires a primary key to be defined.' will be thrown.
Because entityType.GetProperties() only returns scalar properties,  the Id property of  Category  is not included in the returned values of entityType.GetProperties(). I can get the Id property of  Category  with entityType.GetNavigations(),  but IMutableNavigation cannot invoke the SetValueConverter method.

.NET Version:.NET 5
EF Core version:5.0.9

My questioin is :how can I set ValueConverter for non-scalar property at runtime?
Thanks.

**Comment by AndriySvyryd** [comment]
Duplicate of https://github.com/dotnet/efcore/issues/13947

**Comment by AndriySvyryd** [comment]
In 6.0.0-rc1 we'll ship https://github.com/dotnet/efcore/issues/25084 which will allow you to call `.Scalars<CategoryId>().HaveConversion<CategoryIdConverter>()`

**Comment by yangzhongke** [comment]
> In 6.0.0-preview6 we shipped #10784 which allows you to call `.Properties<CategoryId>().HaveConversion<CategoryIdConverter>()`

Thanks for your replies. However,  "Properties<CategoryId>().HaveConversion<CategoryIdConverter>()" cannot solve my problem, because what I want to do is to SetValueConverter at runtime.
```csharp
foreach (var navigation in entityType.GetNavigations())
{
	Type propertyType = navigation.ClrType;
	if (IsStronglyTypedId(propertyType))
	{
		Type valueType = typeof(Guid);
		var converter = StronglyTypedIdConverters.GetOrAdd(
			propertyType,
			_ => CreateStronglyTypedIdConverter(propertyType, valueType));
		**navigation.SetValueConverter(converter);//error: no such method**
	}
}
```
